frontend/src/entities/configuration/dynamic-configuration.tsx

```diff
@@ -218,413 +223,520 @@ export function DynamicConfiguration () {
         setFields({ 'configuration.EMRConfig.applications': updatedSelectedApps });
     };
 
-    return (
-        <Container
-            header={
-                <Header
-                    variant='h2'
-                    description={`The current dynamic configuration of ${window.env.APPLICATION_NAME}. These settings can be modified without redeploying the application.`}
-                >
-                    {window.env.APPLICATION_NAME} Dynamic Configuration
-                </Header>
+
+    const isObject = (x) => typeof x === 'object' && !Array.isArray(x) && x !== null;
+
+    /**
+     * Computes the difference between two JSON objects, recursively.
+     *
+     * This function takes two JSON objects as input and returns a new object that
+     * contains the differences between the two. Works with nested objects.
+     *
+     * @param {object} [obj1={}] - The first JSON object to compare.
+     * @param {object} [obj2={}] - The second JSON object to compare.
+     * @returns {object} - A new object containing the differences between the two input objects.
+     */
+    function getJsonDifference (obj1 = {}, obj2 = {}) {
+        const output = {},
+            merged = { ...obj1, ...obj2 }; // has properties of both
+
+        for (const key in merged) {
+            const value1 = obj1[key], value2 = obj2[key];
+
+            if (isObject(value1) || isObject(value2)) {
+                const value = getJsonDifference(value1, value2); // recursively call
+                if (Object.keys(value).length !== 0) {
+                    output[key] = value;
+                }
+
+            } else {
+                if (JSON.stringify(value1) !== JSON.stringify(value2)) {
```

Comment:
`~.isEqual`may be preferred here: https://lodash.com/docs/4.17.15#isEqual

Reply:
why? `JSON.stringify(value1)` seems to handle all situations we would face. Is there any benefit to using lodash?

Reply:
Generally lodash's libraries are optimized for a variety of edge cases as well as performance.

For primitives it doesn't matter much, but lodash is equal is better in any scenarios where this may be an array or more complex object: https://www.samanthaming.com/tidbits/33-how-to-compare-2-objects/